astropy/io/fits/tests/test_connect.py

```diff
@@ -64,6 +65,15 @@ def setup_class(self):
             dtype=[("a", int), ("b", "U1"), ("c", float)],
         )
 
+    def test_path(self, tmp_path):
+        filename = "temp.fits"
```

Comment:
```suggestion
        filename = tmp_path / "temp.fits"
```

Reply:
I was going to comment about that, indeed....
I am unable to reproduce the error on main using `tmp_path` ... I cannot understand why....

Reply:
Hmm. `tmp_path` might differ from run to run but should remain the same if called twice in a row in a test function. Let me double check locally.

Reply:
> Let me double check locally.

Yes please... I'm starting going mad about that....

Reply:
Note to self: `tmp_path / "temp.fits"` produces `<class 'pathlib.PosixPath'>`

Reply:
@cmarmo , what do you think of https://github.com/cmarmo/astropy/pull/7 ?